onert-micro/onert-micro/src/core/train/OMCheckpointSaver.cpp

```diff
@@ -0,0 +1,433 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "core/OMDataType.h"
+#include "core/train/OMCheckpointSaver.h"
+
+using namespace onert_micro::core::train;
+using namespace onert_micro::core;
+using namespace onert_micro::train;
+using namespace onert_micro;
+
+namespace
+{
+
+// Layers with trainable weights
+// Note: needed not to store some layers with const intputs but it is not trainable (for example
+// Reshape)
+bool isTrainableWeights(const circle::OperatorCode *opcode)
+{
+  switch (opcode->builtin_code())
+  {
+    case circle::BuiltinOperator_FULLY_CONNECTED:
+    case circle::BuiltinOperator_CONV_2D:
+      return true;
+    default:
+      return false;
+  }
+}
+
+constexpr uint16_t MAGIC_NUMBER = 429;
+constexpr uint8_t SCHEMA_VERSION = 1;
+
+} // namespace
+
+/**
+ * Calculate result buffer size
+ **/
+size_t OMCheckpointSaver::calculateFileSize(core::OMRuntimeContext &context,
+                                            OMTrainingStorage &train_storage,
+                                            const OMConfig &config)
+{
+  size_t result = 0;
+
+  // 2 bytes for Magic Number
+  result += 2;
```

Comment:
as a TODO, introduce or use predefined constants of offsets, like as `WOFFieldsOffsets`